src/DynamoCore/Graph/Nodes/NodeModel.cs

```diff
@@ -748,6 +773,16 @@ public virtual void OnNodeModified(bool forceExecute = false)
             if (handler != null) handler(this);
         }
 
+        public event Action<NodeModel> ToggleNodeFreeze;
+        public virtual void OnToggleNodeFreeze()
+        {
+            if (!RaisesModificationEvents)
+                return;
+
+            var handler = ToggleNodeFreeze;
+            if (handler != null) handler(this);
+        }
```

Comment:
this is required to bubble up the event to UI. Otherwise, when you click freeze, the freeze will not be checked. Also, this I need this for setting up the node and connector display.